src/csharp/Grpc.Core.Api/ChannelBase.cs

```diff
@@ -48,5 +49,34 @@ public string Target
         /// </summary>
         /// <returns>A new <see cref="CallInvoker"/>.</returns>
         public abstract CallInvoker CreateCallInvoker();
+
+        /// <summary>
+        /// Shuts down the channel cleanly. It is strongly recommended to shutdown
+        /// the channel once you stopped using it.
+        /// </summary>
+        /// <remarks>
+        /// Guidance for implementors:
+        /// This method doesn't wait for all calls on this channel to finish (nor does
+        /// it have to explicitly cancel all outstanding calls). It is user's responsibility to make sure
+        /// all the calls on this channel have finished (successfully or with an error)
+        /// before shutting down the channel to ensure channel shutdown won't impact
+        /// the outcome of those remote calls.
+        /// </remarks>
+        public Task ShutdownAsync()
+        {
+            return ShutdownAsyncCore();
+        }
+
+        /// <summary>Provides implementation of a non-virtual public member.</summary>
+        #pragma warning disable 1998
+        protected virtual async Task ShutdownAsyncCore()
```

Comment:
Remove async and return `Task.CompletedTask`. Then you don't need to disable the warning.

Reply:
Normally, I'd do that but Task.CompletedTask is not available on net45, so I'd need to do this:
https://github.com/grpc/grpc/blob/8eea254922fb86ab4172e99d851175bc21a2887a/src/csharp/Grpc.Core/Utils/TaskUtils.cs#L36 and it felt wrong just copying the code.
The empty state machine is just a tiny overhead compared to creation and shutting down of a channel (which doesn't happen that often). Also all existing subclasses will soon override this method (GrpcChannel from Grpc.Net.Client will call Dispose() ) and the overhead won't matter anyway.